cmd/restic/backup_progress.go

```diff
@@ -0,0 +1,176 @@
+	ArchiveProgressDoneTemplate     = `\nduration {{duration .Duration}}, {{rate .TotalBytes .Duration}}\n`
+)
+
+var scanProgressUpdateTemplate,
+	scanProgressDoneTemplate,
+	archiveProgressTemplate,
+	archiveProgresEstimateTemplate,
+	archiveProgressDoneTemplate *template.Template
+
+func init() {
+	templateFunctions := template.FuncMap{
+		"duration": formatDuration,
+		"percent":  formatPercent,
+		"bytes":    formatBytes,
+		"seconds":  formatSeconds,
+		"rate":     formatRate,
+	}
+
+	scanProgressUpdateTemplate, _ = template.New("scanProgressUpdateTemplate").Funcs(templateFunctions).Parse(ScanProgressUpdateTemplate)
+	scanProgressDoneTemplate, _ = template.New("scanProgressDoneTemplate").Funcs(templateFunctions).Parse(ScanProgressDoneTemplate)
+	archiveProgressTemplate, _ = template.New("archiveProgressTemplate").Funcs(templateFunctions).Parse(ArchiveProgressUpdateTemplate)
+	archiveProgresEstimateTemplate, _ = template.New("archiveProgressEstimateTemplate").Funcs(templateFunctions).Parse(ArchiveProgressEstimateTemplate)
+	archiveProgressDoneTemplate, _ = template.New("archiveProgressTemplate").Funcs(templateFunctions).Parse(ArchiveProgressDoneTemplate)
+}
+
+// NewProgressStatus return ProgressStatus struct from scanner functions
+func NewProgressStatus(Duration time.Duration, Directories, Files, TotalBytes uint64) *ProgressStatus {
+	return &ProgressStatus{
+		ProgressType: ProgressTypeScanner,
+		Duration:     Duration,
+		Directories:  Directories,
+		Files:        Files,
+		TotalBytes:   TotalBytes,
+	}
+}
+
+// UpdateScanStatus return ProgressStatus struct from scanner functions
+func (ps *ProgressStatus) UpdateScanStatus(Duration time.Duration, Directories, Files, TotalBytes uint64) {
+	ps.ProgressType = ProgressTypeScanner
+	ps.Duration = Duration
+	ps.Directories = Directories
+	ps.Files = Files
+	ps.TotalBytes = TotalBytes
+	ps.Seconds = uint64(Duration / time.Second)
+}
+
+// UpdateProgressStatus update current progress struct with recent data
+func (ps *ProgressStatus) UpdateProgressStatus(Duration time.Duration,
+	Completed, Total, Bps, CompletedBytes uint64,
+	TotalBytes, ItemsDone, ItemsTotal, ETA, Errors uint64) {
+
+	ps.ProgressType = ProgressTypeArchive
+	ps.Duration = Duration
+
+	ps.Completed = Completed
+	ps.Total = Total
+	ps.Bps = Bps
+
+	ps.CompletedBytes = CompletedBytes
+
+	ps.TotalBytes = TotalBytes
+	ps.ItemsDone = ItemsDone
+	ps.ItemsTotal = ItemsTotal
+	ps.Seconds = uint64(Duration / time.Second)
+}
+
+// PrintJSON output of current status
+func (ps *ProgressStatus) PrintJSON() {
+	js, err := json.Marshal(ps)
+
+	if err != nil {
+		msg := fmt.Sprintf("%s\n", err)
+		Warnf(msg)
+		Exitf(100, msg)
+	}
+
+	f := bufio.NewWriter(os.Stdout)
+	fmt.Fprintf(f, "%s\n", js)
+	f.Flush()
+}
+
+// PrintScannerProgress print the scanner content
+func (ps *ProgressStatus) PrintScannerProgress() {
+	var result bytes.Buffer
+	_ = scanProgressUpdateTemplate.Execute(&result, ps)
```

Comment:
Don't just throw away the error here: If you're sure that the error never occurs, add something like `if err != nil { panic(err) }` so that we notice when it somehow becomes non-nil.